Controllers/StockController.cs

```diff
@@ -0,0 +1,282 @@
+    [ResponseCache(VaryByHeader = "User-Agent", Duration = 30)]
+    public class StockController : ControllerBase
+    {
+        private const string AlphaVantageKey = "HNAFS0NKXJHEG5YI";
+        private const string AlphaVantageBaseUrl = "https://www.alphavantage.co/query";
+        private const string FinnhubKey = "btt5puv48v6q0kg1m610";
+        private const string FinnhubBaseUrl = "https://finnhub.io/api/v1";
+        private readonly HttpClient alphaVantageClient;
+        private readonly HttpClient finnhubClient;
+        private readonly JsonSerializerOptions serializerOptions;
+
+        public StockController()
+        {
+            // Configure HttpClients
+            alphaVantageClient = new HttpClient();
+            finnhubClient = new HttpClient();
+            finnhubClient.DefaultRequestHeaders.Add("X-Finnhub-Token", FinnhubKey);
+
+            // Configure JsonSerializer
+            serializerOptions = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                Converters =
+                {
+                    new Int32Converter(),
+                    new LongConverter(),
+                    new DecimalConverter(),
+                    new DateTimeOffsetConverter()
+                }
+            };
+        }
+
+        [HttpGet]
+        public ActionResult Symbols(string exchange)
+        {
+            try
+            {
+                var parameters = $"https://finnhub.io/api/v1/stock/symbol?exchange={exchange.ToUpperInvariant()}";
+
+                HttpResponseMessage response = finnhubClient.GetAsync(parameters).Result;
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception($"Status code: {response.StatusCode}");
+                }
+
+                List<StockSymbol> stockSymbol =
+                    response.Content.ReadFromJsonAsync<List<StockSymbol>>(serializerOptions).Result;
+
+                return Ok(stockSymbol);
+            }
+            catch (Exception e)
+            {
+                return BadRequest($"Request failed. {e.Message}");
+            }
+        }
+
+        [HttpGet("{symbol}/profile")]
+        public ActionResult Profile(string symbol)
+        {
+            try
+            {
+                var parameters = $"https://finnhub.io/api/v1/stock/profile2?symbol={symbol.ToUpperInvariant()}";
+
+                HttpResponseMessage response = finnhubClient.GetAsync(parameters).Result;
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception($"Status code: {response.StatusCode}");
+                }
+
+                CompanyProfile2 profile = response.Content.ReadFromJsonAsync<CompanyProfile2>(serializerOptions).Result;
+
+                if (profile?.Ticker is null)
+                {
+                    throw new Exception("Symbol is invalid or profile does not exist.");
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/InvestingOak/InvestingOak/pullRequest?prid=6380270)